internal/controller/monovertexrollout/monovertexrollout_progressive.go

```diff
@@ -39,43 +44,44 @@ func (r *MonoVertexRolloutReconciler) CreateUpgradingChildDefinition(ctx context
+func (r *MonoVertexRolloutReconciler) AssessUpgradingChild(ctx context.Context, rolloutObject progressive.ProgressiveRolloutObject, existingUpgradingChildDef *unstructured.Unstructured) (apiv1.AssessmentResult, error) {
 	// TODO: Create AnalysisRun for assessing the upgrading child if user creates an AnalysisTemplate and references it in their MonoVertexRollout
-	// The following code can serve as a template for what should work:
-	/*analysisRun := &argorolloutsv1.AnalysisRun{}
-	if err := r.client.Get(ctx, client.ObjectKey{Name: existingUpgradingChildDef.GetName(), Namespace: existingUpgradingChildDef.GetNamespace()}, analysisRun); err != nil {
-		if apierrors.IsNotFound(err) {
-			analysisRun := &argorolloutsv1.AnalysisRun{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      existingUpgradingChildDef.GetName(),
-					Namespace: existingUpgradingChildDef.GetNamespace(),
-					OwnerReferences: []metav1.OwnerReference{
-						*metav1.NewControllerRef(existingUpgradingChildDef, numaflowv1.MonoVertexGroupVersionKind),
-					},
-				},
-				Spec: argorolloutsv1.AnalysisRunSpec{
-					Metrics: []argorolloutsv1.Metric{
-						{
-							Name: "my-metric",
-							Provider: argorolloutsv1.MetricProvider{
-								Prometheus: &argorolloutsv1.PrometheusMetric{
-									Address: "http://prometheus.addon-metricset-ns.svc.cluster.local:9090",
-									Query:   "vector(1) == vector(2)",
-								},
-							},
-						},
-					},
-				},
-			}
-			if err = r.client.Create(ctx, analysisRun); err != nil {
+
+	mvtxRollout := rolloutObject.(*apiv1.MonoVertexRollout)
+	analysis := mvtxRollout.GetAnalysis()
+	if len(analysis.Templates) > 0 {
+		analysisRun := &argorolloutsv1.AnalysisRun{}
+		if err := r.client.Get(ctx, client.ObjectKey{Name: existingUpgradingChildDef.GetName(), Namespace: existingUpgradingChildDef.GetNamespace()}, analysisRun); err != nil {
+			if apierrors.IsNotFound(err) {
+				err := progressive.CreateAnalysisRun(ctx, analysis, existingUpgradingChildDef, r.client)
+				if err != nil {
+					return apiv1.AssessmentResultUnknown, err
+				}
+				// analysisRun is created for the first time
+				analysisStatus := mvtxRollout.GetAnalysisStatus()
+				analysisStatus.AnalysisRunName = existingUpgradingChildDef.GetName()
+				timeNow := metav1.NewTime(time.Now())
+				analysisStatus.StartTime = &timeNow
+				mvtxRollout.SetAnalysisStatus(analysisStatus)
+			} else {
 				return apiv1.AssessmentResultUnknown, err
 			}
+		}
+
+		// assess analysisRun status and set endTime and phase if completed
+		analysisStatus := mvtxRollout.GetAnalysisStatus()
+		if analysisRun.Status.Phase.Completed() && analysisStatus.EndTime == nil {
+			analysisStatus.EndTime = analysisRun.Status.CompletedAt
+			analysisStatus.Phase = analysisRun.Status.Phase
+			mvtxRollout.SetAnalysisStatus(analysisStatus)
 		} else {
-			return apiv1.AssessmentResultUnknown, err
+			analysisStatus.Phase = analysisRun.Status.Phase
+			mvtxRollout.SetAnalysisStatus(analysisStatus)
```

Comment:
maybe just in case somehow we failed to write the AnalysisRunName in the first place (i.e. resource version conflict issue), we can do it here just in case?

Also, you can take this out of the if statement since it pertains to both cases:
```
  analysisStatus.AnalysisRunName = existingUpgradingChildDef.GetName()
  analysisStatus.Phase = analysisRun.Status.Phase
  mvtxRollout.SetAnalysisStatus(analysisStatus)
```